intg-denonavr/denon_remote.py

```diff
@@ -0,0 +1,1226 @@
+
+        Called by the integration-API if a command is sent to a configured remote entity.
+
+        :param cmd_id: command
+        :param params: optional command parameters
+        :return: status code of the command request
+        """
+        return await self._handle_command(cmd_id, params)
+
+    @staticmethod
+    def state_from_avr(avr_state: avr.States) -> ucapi.remote.States:
+        """
+        Convert AVR state to UC API remote state.
+
+        :param avr_state: Denon/Marantz AVR state
+        :return: UC API remote state
+        """
+        if avr_state in REMOTE_STATE_MAPPING:
+            return REMOTE_STATE_MAPPING[avr_state]
+        return ucapi.remote.States.UNKNOWN
+
+    def filter_changed_attributes(self, update: dict[str, Any]) -> dict[str, Any]:
+        """
+        Filter the given attributes and return only the changed values.
+
+        :param update: dictionary with attributes.
+        :return: filtered entity attributes containing changed attributes only.
+        """
+        attributes = {}
+
+        if Attributes.STATE in update:
+            state = DenonRemote.state_from_avr(update[Attributes.STATE])
+            attributes = helpers.key_update_helper(Attributes.STATE, state, attributes, self.attributes)
+
+        return attributes
+
+    async def _handle_command(self, cmd_id: str, params: dict[str, Any] | None = None) -> StatusCodes:
+        command = ""
+        if params:
+            command = params.get("command", "")
+
+        if command == "":
+            command = f"remote.{cmd_id}"
+
+        match command:
+            case "remote.send":
+                return await self._denon_media_player.command(media_player.Commands.ON)
+            case "remote.off":
+                return await self._denon_media_player.command(media_player.Commands.OFF)
+            case "remote.toggle":
+                return await self._denon_media_player.command(media_player.Commands.TOGGLE)
+
+        if cmd_id == Commands.SEND_CMD:
+            return await self._denon_media_player.command(command)
+
+        return StatusCodes.NOT_IMPLEMENTED
+
+    @staticmethod
+    def _get_remote_ui_pages():
+        return [
+            DenonRemote._get_main_page(),
+            DenonRemote._get_sound_modes_page(),
+            DenonRemote._get_standby_page(),
+            DenonRemote._get_triggers_page(),
+            DenonRemote._get_dirac_page(),
+            DenonRemote._get_audyssey_page(),
+            DenonRemote._get_channel_levels_page(),
+            DenonRemote._get_eco_page(),
+        ]
+
+    @staticmethod
+    def _get_main_page():
+        return {
+            "page_id": "denon_avr_commands_main",
+            "name": "Main",
+            "grid": {"width": 4, "height": 7},
+            "items": [
+                {
+                    "command": {
+                        "cmd_id": "remote.send",
+                        "params": {"command": CoreCommands.SPEAKER_PRESET_1, "repeat": 1},
```

Comment:
Note: the command object can be reduced:
- `repeat` is optional and can be omitted. Default is 1.
- when using a defined simple command, it can be directly used in `cmd_id`:
```python
"command": { "cmd_id": CoreCommands.SPEAKER_PRESET_1 }
```

Reply:
Only the commands defined in `simple_commands`, plus the native remote-entity commands on, off, toggle, send_cmd, send_cmd_sequence.
Once the integration allows to send all native Denon commands in send_cmd (like `SITV`), they have to use `send_cmd` and the `params` object. Otherwise the validation should be triggered that it's an unknown command.

Reply:
Bah, I didn't see this command before I made changes. I added your suggestion to simplify the structure and support unknown commands. Maybe that should've been in its own PR?

Reply:
Up to you, the remote-entity is one big new feature so it's fine when it is included.

Reply:
I'll leave it in then